<launch>
  <arg name="pkg" value="olelidar"/>

  <arg name="frame_id" default="olelidar"/>
  <arg name="range_min" default="0.1"/>
  <arg name="range_max" default="50"/>
  <arg name="ang_start" default="0"/>
  <arg name="ang_end" default="360"/>
  <arg name="inverted" default="false"/>
  <arg name="timeFromLidar" default="true"/>
  <arg name="poly" default="1"/>
  <node pkg="$(arg pkg)" type="$(arg pkg)_decoder" name="$(arg pkg)_decoder" output="screen">
    <param name="frame_id" type="string" value="$(arg frame_id)"/>
    <param name="range_min" type="double" value="$(arg range_min)"/>
    <param name="range_max" type="double" value="$(arg range_max)"/>
    <param name="ang_start" type="int" value="$(arg ang_start)"/>
    <param name="ang_end" type="int" value="$(arg ang_end)"/>
    <param name="inverted" type="bool" value="$(arg inverted)"/>
    <param name="timeFromLidar" type="bool" value="$(arg timeFromLidar)"/>
    <param name="poly" type="int" value="$(arg poly)"/>
    <remap from="~packet" to="packet"/>
    <remap from="~scan" to="scan"/>

  </node>

</launch>
